dev/io.openliberty.microprofile.reactive.streams.operators30.cdi/src/io/openliberty/microprofile/reactive/streams/operators30/cdi/ReactiveStreamsEngineInjectionExtension.java

```diff
@@ -0,0 +1,31 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.microprofile.reactive.streams.operators30.cdi;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+
+import com.ibm.ws.cdi.CDIServiceUtils;
+import com.ibm.ws.cdi.extension.WebSphereCDIExtension;
+
+import jakarta.enterprise.event.Observes;
+import jakarta.enterprise.inject.spi.AnnotatedType;
+import jakarta.enterprise.inject.spi.BeanManager;
+import jakarta.enterprise.inject.spi.BeforeBeanDiscovery;
+import jakarta.enterprise.inject.spi.Extension;
+
+@Component(service = WebSphereCDIExtension.class, configurationPolicy = ConfigurationPolicy.IGNORE, property = { "service.vendor=IBM" })
+public class ReactiveStreamsEngineInjectionExtension implements Extension, WebSphereCDIExtension {
+
+    public void beforeBeanDiscovery(@Observes BeforeBeanDiscovery bbd, BeanManager bm) {
+        AnnotatedType<ReactiveStreamsEngineProducer> producer = bm.createAnnotatedType(ReactiveStreamsEngineProducer.class);
+        bbd.addAnnotatedType(producer, CDIServiceUtils.getAnnotatedTypeIdentifier(producer, this.getClass()));
+    }
+}
```

Comment:
I _think_ this is correct for the old way of adding a CDI extension. We have the `WebsphereCDIExtension` component, which is also an `Extension`, the service loader file and a beans.xml to prevent scanning.

Reply:
Are you advocating a switch to being a CDIExtensionMetadata impl?

Reply:
No, but it's a while since I've looked at one done this way, so I think this is right, but I'm not 100%.

I think mostly I just get annoyed at the intricacies of adding a CDI extension in a feature.